### ISSUE TITLE
Quarkus 2.7.0: BasicLoggingEnabler requests configuration without set LaunchMode.TEST

### ISSUE THREAD

**Issue by franden** [quarkusio/quarkus#23505]
In our project we have a custom ConfigSource which access MongoDB to read the configuration. The MongoDB connection string will be injected as `ConfigProperty` to the application according to the set `quarkus.profile`.  The connection string is present in the `application.yml` only for the profiles `test` and `dev`, to be able to run tests and dev mode during the development. For real deployments (STAGE, PROD), the connection string is provided during the deployment process. 

Since Quarkus 2.7.0 all our JUnit test can not be executed because the connection string for MongoDB will be requested during the start of a test for `quarkus.profile=prod` which is not available in our `application.yml` files. The reason therefore is that our MongoDbConfigSource will be initialized before the `LaunchMode` ist set to `LaunchMode.TEST`. This is because of this change: 
https://github.com/quarkusio/quarkus/blob/5e6acd94a417d295a26ac149510254fdd14ef8fc/test-framework/junit5/src/main/java/io/quarkus/test/junit/BasicLoggingEnabler.java#L28

The `LaunchMode.TEST` will be set later :
https://github.com/quarkusio/quarkus/blob/6614c3c97bce85093efcf7f9e32aeae70f4c06b9/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java#L636

Please let `BasicLoggingEnabler` run after `LaunchMode` is set to `LaunchMode.TEST`

**Comment by geoand** [comment]
cc @famod

**Comment by famod** [comment]
I'll have a look later. Do you have a small reproducer to play around with?

**Comment by famod** [comment]
As a short term remedy, I suggest disabling `BasicLoggingEnabler` via `src/test/resources/junit-platform.properties`:
```
junit.jupiter.extensions.autodetection.enabled=false
junit.jupiter.testclass.order.default=io.quarkus.test.junit.util.QuarkusTestProfileAwareClassOrderer
```
(the second line is only to retain the orderer)

You should then also apply the exclusion for `quarkus-junit5-properties` as described in the second part of https://quarkus.io/guides/getting-started-testing#testing_different_profiles

If you have to rely on JUnit autodetection (for other extensions), then you're out of luck for now and I'm very sorry about that.

**Comment by franden** [comment]
Thank you for the quick reply and suggestion for short term remedy. 
Here you can find a reproducer: https://github.com/franden/issue-with-profiles-for-test

**Comment by famod** [comment]
I'm currently testing a fix that will most likely end up in 2.7.2, which will be released on 21st of Feb.
